go/sqlmetrics/sql_intercept.go

```diff
@@ -145,7 +145,7 @@ func RegisterDriverWithLabelMaker(d Options, fn LabelMaker) (string, error) {
 				dc := ctx.(*dbCtx)
 
 				if err := emitDuration(
-					fn(dc.query).Merge(
+					fn(name, dc.query).Merge(
```

Comment:
Not a scalable design you see. Imagine from the consumer who is overriding this. You may have to take a switch based on driver name. Others may do so on other basis. Best way is to return the Options struct itself in the function argument. 

Also, don't use d.GetName() for switch case, use d.Name which is the original unmangled name.